spectre-meltdown-checker.sh

```diff
@@ -1217,10 +1198,23 @@ check_variant3()
 	elif [ -z "$msg" ]; then
 		# if msg is empty, sysfs check didn't fill it, rely on our own test
 		if [ "$opt_live" = 1 ]; then
+			# Test if the current host is a Xen PV Dom0 / DomU
+			dmesg_grep 'Booting paravirtualized kernel on Xen$'; ret=$?
```

Comment:
Do you know any other way of telling whether we're running inside a Xen PV domU? I hate having to rely on `dmesg` :(

Reply:
Well, dmesg provides the most reliable detection, but I'll take a look at Linux sources to know what they rely on to detect this

Reply:
(A bit more detail in here: http://www.brendangregg.com/blog/2014-05-09/xen-feature-detection.html)

Reply:
Yep I've stumbled across this page too, correctly detecting a Xen PV seems surprisingly complicated and error-prone. We can enhance this part later.